packages/itwinui-react/src/core/Table/ColumnHeader.tsx

```diff
@@ -47,16 +45,15 @@ export const ColumnHeader = <
   const {
     columnRefs,
     column,
-    index,
     areFiltersSet,
-    hasAnySubRows,
-    headers,
     state,
```

Comment:
Can `state` also be removed? I assume we can get it from `instance`.

Reply:
Would you mind elaborating more on getting `state` from `instance`? Just want to make sure I get it right.

Reply:
I have stored `instance` within the `TableInstanceContext` but there are some points that I'm not sure about. I'm feeling really uneasy about passing `any` its type parameter but I'm currently having a bit of a hard time finding a way to declare/extend generic type `T` in the context. What should I do to assign type appropriately here? Also, should I create a `useContext` hook and wrap provider around a wrapper component so I won't have to pass `instance` as a prop?

Reply:
Similar to https://github.com/iTwin/iTwinUI/pull/2287#discussion_r1792467871, the generic types are not super important. We can use `Record<string, unknown>`.

I'm not sure what you mean by "create a `useContext` hook and wrap provider around a wrapper component". You can remove the `instance` prop and replace it with this:

```js
const instance = React.useContext(TableInstanceContext);
```

Reply:
I have made this change.

Reply:
There is one part of this thread still pending. I said earlier:

> One thing that we'll need to verify is whether `instance` is a stable/memoized object, i.e. it doesn't change on every re-render.

Can you verify this? You can check the source code for `react-table` v7 and also try something like this code locally:

```js
useEffect(() => { console.log("changed", instance) }, [instance])
```

Reply:
I apologize I got sidetracked by the type issue so I totally missed this :(. I have checked the source code (https://github.com/TanStack/table/blob/v7/src/hooks/useTable.js) and seen that they use either `useMemo` or `useCallback` for the instance's properties, which I believe should result in the instance object being memoized. I have also tried your suggestion above and tried something like this to double check.

``` jsx
const prevInstanceRef = React.useRef(instance);

React.useEffect(() => {
  if (prevInstanceRef.current && prevInstanceRef.current !== instance) {
    console.log('Instance object has changed');
  }
  prevInstanceRef.current = instance;
}, [instance]);
```

So, I believe it should be stable.